src/test/java/teammates/common/util/FieldValidatorTest.java

```diff
@@ -589,6 +591,35 @@ public void testGetInvalidityInfoForTimeForVisibilityStartAndResultsPublish_inva
                          visibilityStart, resultsPublish));
     }
 
+    @Test
+    public void testGetInvalidityInfoForTimeForSessionEndAndExtendedDeadlines_valid_returnEmptyString() {
+        Instant sessionEnd = TimeHelperExtension.getInstantHoursOffsetFromNow(-1);
+        Map<String, Instant> extendedDeadlines = new HashMap<>();
+        extendedDeadlines.put("participant@email.com", TimeHelperExtension.getInstantHoursOffsetFromNow(1));
+        assertEquals("",
+                FieldValidator.getInvalidityInfoForTimeForSessionEndAndExtendedDeadlines(
+                        sessionEnd, extendedDeadlines));
+    }
+
+    @Test
+    public void testGetInvalidityInfoForTimeForSessionEndAndExtendedDeadlines_invalid_returnErrorString() {
+        ______TS("extended deadline earlier than the end time");
+        Instant sessionEnd = TimeHelperExtension.getInstantHoursOffsetFromNow(1);
+        Map<String, Instant> extendedDeadlines = new HashMap<>();
+        extendedDeadlines.put("participant@email.com", TimeHelperExtension.getInstantHoursOffsetFromNow(-1));
```

Comment:
I think it should be covered?

I assume the deadlines and the end time are valid at first, so at first, all the deadlines are after the end time. When updating the maps,

#### Case 1: deadlines and end time are both unchanged

All deadlines are guaranteed to be after the end time, so this is covered in the test case with valid inputs.

#### Case 2: deadlines unchanged and end time changed (moved back or front) but still before all deadlines

All deadlines are still after the end time, so this is covered in the test case with valid inputs.

#### Case 3: deadlines unchanged and end time moved forward after some deadlines

Some deadlines are now before the end time, so this is covered in the test case with invalid inputs.
<br>

Unless you mean something else?

Reply:
Oh what I meant was that the deadline and the end time are the same. Can we get a guarantee in this test too?

Reply:
Oh I see. I think I intended to check the deadline being the same as the end time in the second section of the test. In that section, I replace the deadline of the sole participant with the same time as the `sessionEnd`. So I believe I have covered it. Let me know if you think it still does not cover it.

However, since you brought this to my attention, I realised I may not have changed the section description to reflect that after copy-pasting it. So I will make this change. The wrong test section description might be why you did not notice it, so sorry.